server/routerlicious/packages/services-telemetry/src/sanitizationLumberFormatter.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ILumberFormatter, LogLevel } from "./resources";
+import { Lumber } from "./lumber";
+
+export class SanitizationLumberFormatter implements ILumberFormatter {
+	private readonly sensitiveKeys = [
+		/cookie/i,
+		/passw(or)?d/i,
+		/^pw$/,
+		/^pass$/i,
+		/secret/i,
+		/token/i,
+		/api[._-]?key/i,
+		/session[._-]?id/i,
+	];
+
+	private readonly redactedStr = "[LUMBER_REDACTED]";
+
+	public transform(lumber: Lumber<string>): void {
+		if (lumber.logLevel === LogLevel.Error && lumber.exception) {
+			const sensitiveKeys = new Set<string>();
+
+			this.redactException(lumber.exception, sensitiveKeys);
+
+			if (sensitiveKeys.size > 0) {
+				lumber.setProperty("SensitiveDataDetected", sensitiveKeys);
```

Comment:
nit: rename to "detectedSensitiveKeys" to not confuse with the same name class member?

Reply:
Renamed.